test/e2e/README.md

```diff
@@ -66,10 +74,94 @@ uses a matrix strategy to invoke each area as its own job, running in parallel.
 > We're still learning what the best way to arrange E2E tests is. It's okay for
 > this pattern to change if it's not working well, but please discuss first!
 
-Each feature under test consists of:
+We try to follow this pattern when adding a new test:
+
+1. Define a single feature per Test function, if possible.
+1. Setup a directory of plain YAML manifests per test - i.e. test fixtures - at
+   `e2e/manifests/<area>/<test>`, usually with a `prerequisites` sub-folder
+   containing resources to be deployed at setup phase and cleaned up during the
+   teardown. Try to avoid reusing other feature's fixtures, as this would introduce
+   hidden dependencies between tests.
+1. Try reusing existing helpers as much as possible, see package
+   `github.com/crossplane/crossplane/test/e2e/funcs`, or add new ones there if
+   needed.
+1. Prefer using the Fluent APIs to define features
+   (`features.New(...).WithSetup(...).Assess(...).WithTeardown(...).Feature()`).
+   1. `features.Table` should be used only to define multiple self-contained
+      assessments to be run sequentially, but without assuming any ordering among
+      them, similarly to the usual table driven style we adopt for unit testing.
```

Comment:
It's not about the number of assessments, it's about the kind of assessments as we discussed. Table testing can be used when the order of the test cases (that here are assessments) is not relevant and therefore each assessment is self-contained and does not rely nor can be influenced by other assessments.
See the validation test or the example in the e2e folder as an example of an idiomatic table driven test.

Reply:
I see. And to make sure I understand, this isn't because a `features.Table` is actually run in random order in practice but rather because we think folks would _expect_ that behaviour - i.e. expect a table driven test to not be self/contained and not order dependent.

Reply:
Yes, exactly, it's not, but that's what people expect by table driven testing.